yarn-project/archiver/src/archiver/archiver_store.ts

```diff
@@ -299,18 +293,21 @@ export class MemoryArchiverStore implements ArchiverDataStore {
    * @param from - Number of the first block to return (inclusive).
    * @param limit - The number of blocks to return.
    * @returns The requested L2 blocks.
+   * @remarks When "from" is smaller than genesis block number, blocks from the beginning are returned.
    */
-  public getL2Blocks(from: number, limit: number): Promise<L2Block[]> {
+  public getBlocks(from: number, limit: number): Promise<L2Block[]> {
     // Return an empty array if we are outside of range
     if (limit < 1) {
-      throw new Error(`Invalid block range from: ${from}, limit: ${limit}`);
+      throw new Error(`Invalid limit: ${limit}`);
     }
-    if (from < INITIAL_L2_BLOCK_NUM || from > this.l2BlockContexts.length) {
+
+    const fromIndex = Math.max(from - INITIAL_L2_BLOCK_NUM, 0);
```

Comment:
we use `INITIAL_L2_BLOCK_NUM` here but not in the archiver? Is that an issue?

Reply:
It was being used in an overloaded context in the archiver. ( incrementing to next block / getting the first block number )